pennylane_lightning_kokkos/src/simulator/MeasuresFunctors.hpp

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_StdAlgorithms.hpp>
+
+#include "Util.hpp"
+
+namespace {
+using namespace Pennylane::Util;
+namespace KE = Kokkos::Experimental;
+} // namespace
+
+namespace Pennylane {
+namespace Functors {
+
+template <class Precision> struct getProbFunctor {
+
+    Kokkos::View<Kokkos::complex<Precision> *> arr;
+    Kokkos::View<Precision *> probilities;
+
+    getProbFunctor(Kokkos::View<Kokkos::complex<Precision> *> arr_,
+                   Kokkos::View<Precision *> probilities_)
+        : arr(arr_), probilities(probilities_) {}
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const size_t k) const {
+        Precision REAL = arr[k].real();
+        Precision IMAG = arr[k].imag();
+        probilities[k] = REAL * REAL + IMAG * IMAG;
+    }
+};
+
+template <class Precision> struct getCDFFunctor {
+
+    Kokkos::View<Precision *> probilities;
+
+    getCDFFunctor(Kokkos::View<Precision *> probilities_)
+        : probilities(probilities_) {}
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const size_t &k, Precision &update_value,
+                    const bool final) const {
+        const Precision val_k = probilities[k];
+
+        if (final)
```

Comment:
I'm curious, how does `final` as an if-statement condition work?

Reply:
Let me walk through the source code and get back to you.

Reply:
I misread the code --- apologies. We shouldn't use `final` as an argument name, as this is a CPP specifier: https://en.cppreference.com/w/cpp/language/final

Reply:
let me fix it.

Reply:
done